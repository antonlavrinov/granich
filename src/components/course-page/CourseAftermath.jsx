import React from 'react'
import { Container } from '../style';
import styled from 'styled-components';
import BackgroundImage from "gatsby-image";
// import {documentToReactComponents} from '@contentful/rich-text-react-renderer';
import { graphql, useStaticQuery } from 'gatsby'



const AftermathSection = styled.section`
    margin-bottom: 4vw;
    @media only screen and (max-width: 575px) {
        margin-bottom: 5vw;
    }
`

const AftermathSectionTitle = styled.h2`
    font-size: 4vw;
    font-weight: 700;
    letter-spacing: -0.1vw;
    margin-bottom: 2.4vw;
    line-height: 1;
    
    span {
        font-weight: inherit;
        position: relative;
        font-family: EB Garamond;
        font-style: italic;
        font-weight: 500;
        font-size: 5.5vw;
        letter-spacing: -0.2vw;
    }
    @media only screen and (max-width: 575px) {
        font-size: 11vw;
        margin-bottom: 4vw;
        br {
            display: none;
        }
        span {
            font-size: 13vw;
        }
    }
`

const AftermathWrapper = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-column-gap: 1vw;
    grid-row-gap: 1.5vw;
    @media only screen and (max-width: 575px) {
        grid-template-columns: 1fr 1fr;
        grid-row-gap: 3vw;
        grid-column-gap: 3vw;
    }

`

const AftermathBlockWrapper = styled.div`
    background: white;
    padding: 1.8vw;
    border-radius: 0.6vw;
    @media only screen and (max-width: 575px) {
        border-radius: 2vw;
        padding: 3vw 3vw 4vw;
    }

`

const AftermathBlockImage = styled(props => <BackgroundImage {...props}/>)`
    background: url(${props => props.image}) center center no-repeat;
    // background-size: 90%;
    width: 100%;
    height: 15vw;
    border-radius: 0.2vw;
    margin-bottom: 1vw;
    @media only screen and (max-width: 575px) {
        height: 35vw;
        border-radius: 1.5vw;
        margin-bottom: 5vw;
    }
`
const AftermathBlockTextBlock = styled.div`
    line-height: 1.35;
    margin-left: -0.1vw;
    font-size: 1.15vw;
    color: var(--granich-grey);
    letter-spacing: -0.02vw;
    max-width: 15vw;
    span {
        font-weight: 500;
        color: var(--granich-black);
        display: block;
    }
    i {
        font-style: normal;
        display: block;
        color: var(--granich-red);
        font-weight: 500;
    }
    @media only screen and (max-width: 575px) {
        margin-left: 0;
        font-size: 3.7vw;
        max-width: 100%;
        line-height: 1.25;
    }
`



const AftermathBlock = ({text, image }) => {



    return (
        <AftermathBlockWrapper>
            {/* <AftermathBlockTitle>{title}</AftermathBlockTitle> */}
            <AftermathBlockImage fluid={image}/>
            <AftermathBlockTextBlock>
                {/* {documentToReactComponents(text)} */}
                {text}
            </AftermathBlockTextBlock>
        </AftermathBlockWrapper> 
    )
}

const CourseAftermath = ({data}) => {
    const dataImages = useStaticQuery(graphql`
        query aftermathImages {
            aftermathImage1: file(relativePath: { eq: "graph-design/aftermath/aftermath-image-01.jpg" }) {
                childImageSharp {
                    fluid(maxWidth: 350, quality: 90) {
                        ...GatsbyImageSharpFluid
                    }
                }
            }
            aftermathImage2: file(relativePath: { eq: "graph-design/aftermath/aftermath-image-02.jpg" }) {
                childImageSharp {
                    fluid(maxWidth: 350, quality: 90) {
                        ...GatsbyImageSharpFluid
                    }
                }
            }
            aftermathImage3: file(relativePath: { eq: "graph-design/aftermath/aftermath-image-03.jpg" }) {
                childImageSharp {
                    fluid(maxWidth: 350, quality: 90) {
                        ...GatsbyImageSharpFluid
                    }
                }
            }
            aftermathImage4: file(relativePath: { eq: "graph-design/aftermath/aftermath-image-04.jpg" }) {
                childImageSharp {
                    fluid(maxWidth: 350, quality: 90) {
                        ...GatsbyImageSharpFluid
                    }
                }
            }

        }
    `)

    const imageData1 = dataImages.aftermathImage1.childImageSharp.fluid
    const imageData2 = dataImages.aftermathImage2.childImageSharp.fluid
    const imageData3 = dataImages.aftermathImage3.childImageSharp.fluid
    const imageData4 = dataImages.aftermathImage4.childImageSharp.fluid

    return (
        <AftermathSection>
            <Container>
                {/* <AftermathSectionTitle>
                После курса вы получите <span>твёрдую базу</span> для дальнейшего изучения:
                </AftermathSectionTitle> */}
                <AftermathSectionTitle>
                Вы получите <span>твёрдую базу</span> <br/>для дальнейшего изучения:
                </AftermathSectionTitle>
                <AftermathWrapper>
                    {/* {data.edges.map(Aftermath => {
                        return (
                            <AftermathBlock key={Aftermath.node.id} image={Aftermath.node.AftermathImage.fluid}
                                title={Aftermath.node.AftermathTitle}
                                text={Aftermath.node.AftermathText.json}
              
                            />
                        )
                    })} */}
                    <AftermathBlock 
                        image={imageData1}
                        text={(<><span>Медиадизайна</span>промосайты, интерактивные лонгриды. (Медиадизайн — это подраздел веб-дизайна)</>)}
    
                    />
                    <AftermathBlock 
                        image={imageData2}
                        text={(<><span>Айдентики
                            </span>визуальное оформление компаний</>)}
    
                    />
                    <AftermathBlock 
                        image={imageData3}
                        text={(<><span>Дизайна упаковки
                            </span>оформление потребительских товаров</>)}
    
                    />
                    <AftermathBlock 
                        image={imageData4}
                        text={(<><span>UI-дизайна
                            </span>оформление интерфейсов диджитал-продуктов, <i>не путать с UX!</i>
</>)}
    
                    />

                </AftermathWrapper>
            </Container>
        </AftermathSection>
    )
}

export default CourseAftermath
