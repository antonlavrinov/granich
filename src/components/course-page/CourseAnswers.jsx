import React,{memo, useState, useRef, useEffect } from 'react'
import { Container } from '../style';
import styled from 'styled-components';
import {documentToReactComponents} from '@contentful/rich-text-react-renderer';
import ArrowTop from '../../assets/svgs/header-arrow-top.svg';
import ArrowTopMobile from '../../assets/svgs/header-arrow-top_mobile.svg';

import scrollTo from 'gatsby-plugin-smoothscroll';
import { useSpring, a } from 'react-spring';
import ResizeObserver from 'resize-observer-polyfill'
import PlusSign from '../../assets/svgs/plus-sign.svg';
import MinusSign from '../../assets/svgs/minus-sign.svg';
import { animated } from 'react-spring';
import {
    Accordion,
    AccordionItem,
    AccordionItemHeading,
    AccordionItemButton,
    AccordionItemPanel,
} from 'react-accessible-accordion';
import 'react-accessible-accordion/dist/fancy-example.css';


const toggle = {
    cursor: 'pointer',
    verticalAlign: 'middle'
}

//helpers
function usePrevious(value) {
    const ref = useRef()
    useEffect(() => void (ref.current = value), [value])
    return ref.current
  }
  
function useMeasure() {
    const ref = useRef()
    const [bounds, set] = useState({ left: 0, top: 0, width: 0, height: 0 })
    const [ro] = useState(() => new ResizeObserver(([entry]) => set(entry.contentRect)))
    useEffect(() => {
      if (ref.current) ro.observe(ref.current)
      return () => ro.disconnect()
    }, [])
    return [{ ref }, bounds]
  }

const AnswersItem = memo(({ type, children, tag, style, open, defaultOpen = false }) => {
    const [isOpen, setOpen] = useState(open === 0 ? true : defaultOpen)
    const previous = usePrevious(isOpen)
    const [bind, { height: viewHeight }] = useMeasure()
    const { height, opacity, transform } = useSpring({
        from: { height: 0, opacity: 0, transform: 'translate3d(20px,0,0)' },
        to: { height: isOpen ? viewHeight : 0, opacity: isOpen ? 1 : 0, transform: `translate3d(${isOpen ? 0 : 20}px,0,0)` }
    })
    return (
        <AnswersItemFrame type={type ? 1 : 0}>
            <AnswersContainer type={type ? 1 : 0} onClick={() => setOpen(!isOpen)}>
                <AnswersTagColumn>
                    <AnswersTag type={type ? 1 : 0}>{tag}</AnswersTag>
                </AnswersTagColumn>
                <AnswersLine type={type ? 1 : 0}/>
                {!isOpen ? (
                    <Plus style={{ ...toggle, opacity: children ? 1 : 0.3 }} />
                ) : (
                    <Minus style={{ ...toggle, opacity: children ? 1 : 0.3 }}/>
                )}
            </AnswersContainer>

            <AnswersItemContent style={{ opacity, height: isOpen && previous === isOpen ? 'auto' : height }}>
                <a.div style={{ transform }} {...bind} children={children} />
                
            </AnswersItemContent>

        </AnswersItemFrame>

    )
})


export const AnswersLesson = styled.div`
    font-size: 1.15vw;
    border: 1px solid var(--granich-red);
    color: var(--granich-red);
    border-radius: 100vw;
    padding: 0.4vw 0;
    font-weight: 500;
    display: inline-block;
    min-width: 5.5vw;
    text-align: center;
    user-select: none;
    ${props => !props.type && `
        background: var(--granich-red);
        color: white;
    `}
    @media only screen and (max-width: 575px) {

        padding: 1vw 0;
        min-width: 15vw;
        font-size: 3.3vw;

    }
`



export const AnswersLessonColumn = styled.div`
    width: 6.8vw;
    @media only screen and (max-width: 575px) {
        width: 20vw;
    }
`

const AnswersLine = styled.div`
    width: 78%;
    height: 1px;
    border-bottom: 1.5px solid black;
    margin-left: auto;
    margin-right: 2v;
    ${props => props.type && `
        border-bottom: 1px solid var(--granich-light-grey);
    `}
`

const Plus = styled(props => <PlusSign {...props}/>)`
    width: 2.3vw;
    height: 2.3vw;
    margin-left: auto;
`
const Minus = styled(props => <MinusSign {...props}/>)`
    width: 2.3vw;
    height: 2.3vw;
    margin-left: auto;
`

const AnswersItemFrame = styled.div`
    position: relative;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow-x: hidden;
    vertical-align: middle;
    fill: white;
    overflow: hidden;
`

const AnswersContainer = styled.div`
    display: flex;
    padding: 1.7vw 0vw;
    align-items: center;
    :hover {
        cursor: pointer;
    }
`

const AnswersTag = styled.div`
    font-size: 1.15vw;
    border: 1.5px solid var(--granich-red);
    color: var(--granich-red);
    border-radius: 100vw;
    padding: 0.3vw 0;
    font-weight: 400;
    display: inline-block;
    min-width: 9vw;
    text-align: center;
    user-select: none;
    ${props => !props.type && `
        background: var(--granich-red-gradient);
        border: none;
        padding: 0.4vw 0;
        color: white;
        box-shadow: 0.1vw 0.1vw 0.3vw rgba(0,0,0,0.3);
    `}
    @media only screen and (max-width: 575px) {

        padding: 1vw 0;
        min-width: 25vw;
        font-size: 3.3vw;
        border: 1px solid var(--granich-red);

    }
`



const AnswersTagColumn = styled.div`
    width: 9.8vw;
    @media only screen and (max-width: 575px) {
        width: 20vw;
    }
`


const AnswersItemContent = styled(animated.div)`
    will-change: transform, opacity, height;
    overflow: hidden;

`



const ArrowLinkTop = styled(props => <ArrowTop {...props}/>)`
  position: absolute;
  top: -0.1vw;
  right: -1.4vw;
  width: 1vw;
  height: 1vw;
  @media only screen and (max-width: 575px) {
    top: -1.6vw;
    right: -3.7vw;
    width: 3.1vw;
    height: 3.1vw;
  }


`
const ArrowLinkTopMobile = styled(props => <ArrowTopMobile {...props}/>)`
  position: absolute;
  @media only screen and (max-width: 575px) {
    top: -1.6vw;
    right: -3.7vw;
    width: 3.1vw;
    height: 3.1vw;
  }


`

const AnswersSection = styled.section`

    margin-bottom: 4vw;
    margin-top: 4vw;
    @media only screen and (max-width: 575px) {
        margin-bottom: 5vw;
        margin-top: 5vw;

    }
` 

const AnswersMainWrapper = styled.div`
    background: #E8E8E8;
    border-radius: 0.6vw;
    @media only screen and (max-width: 575px) {
        border-radius: 2vw;
    }
`

const AnswersWrapper = styled.div`

    padding: 5.4vw 3vw 2vw;
    border-bottom: 0.25vw dotted white;
    @media only screen and (max-width: 575px) {
        padding: 5.4vw 5vw 5.4vw;
        border-bottom: 0.8vw dotted white;
    }

`

const AnswersHeader = styled.div`
    display: flex;
    padding-bottom: 5.7vw;
    // border-bottom: 1.5px solid var(--granich-black);
    width: 100%;
    @media only screen and (max-width: 575px) {
        flex-direction: column;
    }
`

const AnswersSectionTitle = styled.div`
    color: var(--granich-black);
    font-size: 4.55vw;
    letter-spacing: -0.2vw;
    font-weight: 700;
    line-height: 0.85;
    margin-right: 2vw;
    width: 40vw;
    min-width: 40vw;
    @media only screen and (max-width: 575px) {
        font-size: 11vw;
        letter-spacing: -0.7vw;
        line-height: 1;
        margin-right: 0;
        margin-bottom: 4vw;
        width: 100%;
    }

`

const AnswersSectionText = styled.div`
    color: var(--granich-black);
    font-size: 1.5vw;
    margin-top: 0.2vw;
    line-height: 1.55;
    letter-spacing: 0.025vw;
    font-weight: 500;
    // width: 45vw;
    @media only screen and (max-width: 575px) {
        font-size: 3.9vw;
        letter-spacing: -0.08vw;
    }
    

`





const AnswersTextWrapper = styled.div`
    display: flex;
`

const AnswerCategoryColumn = styled.div`
    width: 27vw;
    min-width: 27vw;
    margin-right: 2vw;
    line-height: 1.45;
    p {
        font-size: 1.12vw;
        color: var(--granich-grey);
        margin-bottom: 0.5vw;
        display: block;
        white-space: normal;
    }
    b {
        display: block;
        font-weight: 500;
        margin-top: 1vw;
    }
    @media only screen and (max-width: 575px) {
        width: 100%;
        min-width: 100%;
        padding: 0;
        margin-right: 0;   
        p {
            min-width: 100%;
            margin-bottom: 3vw;
            font-size: 3.7vw;
            line-height: 1.45;
        }
    }

`

const AnswersFooterContainer = styled.div`
    padding: 3vw 4.8vw 5vw;
    @media only screen and (max-width: 575px) {
        padding: 6vw 6vw 8vw;
    }
`

const AnswersFooterWrapper = styled.div`
    background: #dedede;
    border-radius: 0.5vw;
    font-family: EB Garamond;
    font-style: italic;
    font-size: 2.37vw;
    font-weight: 500;
    text-align: center;
    padding: 1.4vw;
    letter-spacing: -0.02vw;
    line-height: 1;
    display: block;
    @media only screen and (max-width: 575px) {
        display: none;
    }
`

const AnswersFooterWrapperMobile = styled.div`
    background: #dedede;
    font-family: EB Garamond;
    font-style: italic;
    font-weight: 500;
    letter-spacing: -0.02vw;
    line-height: 1;
    display: none;
    @media only screen and (max-width: 575px) {
        font-size: 6.6vw;
        padding: 4vw 6vw 6vw;
        line-height: 1.1;
        border-radius: 1.5vw;
        text-align: center;
        display: block;
    }
`
const AnswersFooterLink = styled.a`
    font-style: italic;
    font-family: EB Garamond;
    font-size: 2.37vw;
    font-weight: 500;
    display: inline-block;
    line-height: 1;
    position: relative;
    text-align: center;
    :before {
        content: '';
        position: absolute;
        left: 0;
        bottom: -0.5vw;
        width: 100%;
        height: 1.5px;
        background: #C0C0C0;
    }
    :hover {
        cursor: pointer;
        :before {
            display: none;
        }
    }
    @media only screen and (max-width: 575px) {
        font-size: 6.6vw;
        :before {
            bottom: -2vw;

        }
    }
`


export const AnswersContentWrapper = styled.div`

    display: flex;
    align-items: flex-start;
    padding-left: 11.3vw;
    padding-top: 0vw;
    padding-bottom: 1.2vw;
`

export const AnswersAccordeon = styled.div`
    display: block;  
    @media only screen and (max-width: 575px) {
        display: none;
    }
`

export const EmptyText = styled.div`
    opacity: 0;
    font-size: 0;
`

export const AnswersAccordionMobile = styled(props => <Accordion {...props}/>)`
    display: none;  
    @media only screen and (max-width: 575px) {
        display: block;
        margin-top: 5vw;
    }
`
export const AnswersAccordionItem = styled(props => <AccordionItem {...props}/>)`
 margin: 3vw 0;
 padding: 2.5vw 0 0;
 cursor: pointer;
`

export const AnswersAccordionHeading = styled(props => <AccordionItemHeading {...props}/>)`
    position: relative;
`
export const AnswersAccordionButton = styled(props => <AccordionItemButton {...props}/>)`
    padding: 3vw 2vw;
    user-select: none;
    position: relative;
    :hover {
        cursor: pointer;
    }
    :focus {
        outline: none;
    }
`
export const AnswersLessonAndLineWrapper = styled.div`
    display: flex;
    align-items: center;
`

export const AnswersItemLineMobile = styled.div`
    width: 54.5%;
    height: 1px;
    background: var(--granich-light-grey);
    margin-left: auto;
    margin-right: 8vw;
`




const CourseAnswers = ({data}) => {
    return (
        <AnswersSection>
            <Container>
                <AnswersMainWrapper>
                    <AnswersWrapper>
                        <AnswersHeader>
                            <AnswersSectionTitle>Точечные ответы о курсе</AnswersSectionTitle>
                            <AnswersSectionText>Тут мы собрали ответы на самые популярные вопросы о курсе Осознанный графдизайн</AnswersSectionText>
                        </AnswersHeader>
                        <AnswersAccordeon>
                            {data.edges.map((tab, idx) => {
                                return (
                                    <AnswersItem open={idx} key={tab.node.id} tag={tab.node.answersTagName} type={tab.node.answersTagType ? 1 : 0}>
                                        <AnswersContentWrapper>
                                        <AnswersTextWrapper>
                                            {tab.node.childContentfulGranichCourseAnswersAnswersFirstColumnRichTextNode && <AnswerCategoryColumn><EmptyText>sometext</EmptyText>{documentToReactComponents(tab.node.childContentfulGranichCourseAnswersAnswersFirstColumnRichTextNode.json)}</AnswerCategoryColumn>}
                                            {tab.node.childContentfulGranichCourseAnswersAnswersSecondColumnRichTextNode && <AnswerCategoryColumn>{documentToReactComponents(tab.node.childContentfulGranichCourseAnswersAnswersSecondColumnRichTextNode.json)}</AnswerCategoryColumn>} 
                                        </AnswersTextWrapper>

                                    
                                        </AnswersContentWrapper>
                                    </AnswersItem>
                                )
                            })}
                        </AnswersAccordeon>
                        <AnswersAccordionMobile preExpanded={[0]} allowMultipleExpanded allowZeroExpanded>
                        {data.edges.map((tab, idx) => {
                            const type = tab.node.answersTagType;
                            const tag = tab.node.answersTagName;
                            return (
                                <AnswersAccordionItem uuid={idx} key={tab.node.id}>
                                    <AnswersAccordionHeading>
                                        <AccordionItemButton className="accordion-answers-button">
                                            <AnswersLessonAndLineWrapper>
                                                <AnswersTagColumn>
                                                    <AnswersTag type={type ? 1 : 0}>{tag}</AnswersTag>
                                                </AnswersTagColumn>
                                                <AnswersItemLineMobile/>
                                            </AnswersLessonAndLineWrapper>                                         
                                        </AccordionItemButton>
                                    </AnswersAccordionHeading>
                                    <AccordionItemPanel className="accordion__panel accordion__panel_answers">
                                            {tab.node.childContentfulGranichCourseAnswersAnswersFirstColumnRichTextNode && <AnswerCategoryColumn><EmptyText>sometext</EmptyText>{documentToReactComponents(tab.node.childContentfulGranichCourseAnswersAnswersFirstColumnRichTextNode.json)}</AnswerCategoryColumn>}
                                            {tab.node.childContentfulGranichCourseAnswersAnswersSecondColumnRichTextNode && <AnswerCategoryColumn>{documentToReactComponents(tab.node.childContentfulGranichCourseAnswersAnswersSecondColumnRichTextNode.json)}</AnswerCategoryColumn>} 
                                    </AccordionItemPanel>
                                </AnswersAccordionItem>
                            )
                        } )}

                        </AnswersAccordionMobile>


                    </AnswersWrapper>
                    <AnswersFooterContainer>
                        <AnswersFooterWrapper>
                            <AnswersFooterLink onClick={() => scrollTo('#participation-section')}>Вернуться к оплате Курса!<ArrowLinkTop/></AnswersFooterLink>
                        </AnswersFooterWrapper>
                        <AnswersFooterWrapperMobile>
                            Вернуться <br/>к оплате <AnswersFooterLink onClick={() => scrollTo('#participation-section')}>Курса!<ArrowLinkTopMobile/></AnswersFooterLink>
                        </AnswersFooterWrapperMobile>
                    </AnswersFooterContainer>


                </AnswersMainWrapper>
            </Container>

        </AnswersSection>
    )
}

export default CourseAnswers
