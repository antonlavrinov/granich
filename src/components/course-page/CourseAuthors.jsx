import React from 'react'
import {TeacherBlock} from '../OurTeam';
import { Container } from '../style';
import styled from 'styled-components';

const AuthorsSection = styled.section`
    margin-bottom: 4vw;
    @media only screen and (max-width: 575px) {
        margin-bottom: 5vw;
    }
`

const AuthorsWrapper = styled.section`
 padding: 4.7vw 7vw 4.7vw 4.7vw;
 border-radius: 0.6vw;
 background: white;
 @media only screen and (max-width: 575px) {
    padding: 6vw;
    border-radius: 3vw;
 }
`

const AuthorsTitle = styled.div`
    color: var(--granich-black);
    font-size: 4.55vw;
    letter-spacing: -0.2vw;
    font-weight: 700;
    line-height: 0.87;
    margin-top: -0.6vw;
    margin-bottom: 2.3vw;

    @media only screen and (max-width: 575px) {
        font-size: 11.3vw;
        letter-spacing: -0.7vw;
        line-height: 1;
        margin-right: 0;
        margin-bottom: 2vw;
    }
`

const AuthorsTeachers = styled.div`
    display: flex;
    margin-bottom: 2.5vw;
    @media only screen and (max-width: 575px) {
        flex-direction: column;
        margin-bottom: 0vw;
    }

`

const TeacherWrapper = styled.div`

    position: relative;
    :first-child {
        margin-right: 5vw;
        :after {
            content: '+';
            position: absolute;
            color: var(--granich-black);
            font-weight: 600;
            top: 50%;
            transform: translateY(-50%);
            right: -4.2vw;
            font-size: 5vw;

        }
    }
    @media only screen and (max-width: 575px) {

        :first-child {
            margin-right: 0;
            margin-bottom: 3vw;
            :after {
                right: auto;
                transform: translate(-50%, 0);
                left: 50%;
                bottom: -8vw;
                top: auto;
                font-size: 13vw;
    
            }
    }
    `
const AuthorsText = styled.div`
    display: flex;
    @media only screen and (max-width: 575px) {
        flex-direction: column;
    }
`
const AuthorsQuote = styled.div`
    font-size: 1.55vw;
    color: var(--granich-black);
    padding-left: 1.5vw;
    border-left: 2.5px solid var(--granich-red);
    margin-right: 1vw;
    font-weight: 500;
    min-width: 28vw;
    line-height: 1.45;
    @media only screen and (max-width: 575px) {
        font-size: 5.1vw;
        letter-spacing: -0.1vw;
        padding-left: 3.5vw;
        margin-right: 0;
        margin-bottom: 3vw;
    }
`
const AuthorsDescr = styled.div`
    font-size: 1.15vw;
    color: var(--granich-grey);
    padding-top: 0.5vw;
    line-height: 1.45;
    @media only screen and (max-width: 575px) {
        font-size: 3.8vw;
    }
`

const CourseAuthors = ({data}) => {
    return (
        <AuthorsSection>
            <Container>
                <AuthorsWrapper>
                    <AuthorsTitle>Соавторы</AuthorsTitle>
                    <AuthorsTeachers>
                        {data.edges.map((teacher) => {
                            return (
                                <TeacherWrapper key={teacher.node.id}>
                                    <TeacherBlock masterClass teacher={teacher.node}/>
                                </TeacherWrapper>

                            )
                        })}
                    </AuthorsTeachers>
                    <AuthorsText>
                        <AuthorsQuote>
                            Мастер-класс сделан совместно с Елизаветой Черниковой, видеолекции и инструкции записывала она
                        </AuthorsQuote>
                        <AuthorsDescr>Елизавета прошла у меня суровую подготовку, знает все необходимые стандарты качества, ценит осознанный подход. Именно она является главным исследователем в моей команде. Вдобавок, она, вместе со мной, ведет Энциклопедию Графдизайна на Пинтересте.</AuthorsDescr>
                    </AuthorsText>

                    
                </AuthorsWrapper>
            </Container>
            
        </AuthorsSection>
    )
}

export default CourseAuthors
