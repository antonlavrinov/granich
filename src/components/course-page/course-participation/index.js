import React from "react";
import { Container } from "../../style";
import ParticipationForm from "../participation/form";
import ParticipationPolicy from "../participation/ParticipationPolicy";
import * as SC from "./style";

const CourseParticipation = ({
  data,
  formId,
  formAction,
  googleAnaliticsCategory,
  additionalTags,
  telegram,
  additionalComponent,
  policy,
  individualSupport,
}) => (
  <SC.ParticipationSection>
    <Container>
      <SC.ParticipationWrapper policy={policy}>
        <ParticipationForm
          additionalTags={additionalTags}
          googleAnaliticsCategory={googleAnaliticsCategory}
          formId={formId}
          formAction={formAction}
          data={data}
          policy={policy}
          individualSupport={individualSupport}
        />
        {policy && (
          <ParticipationPolicy
            additionalComponent={additionalComponent}
            telegram={telegram}
            data={data}
          />
        )}
      </SC.ParticipationWrapper>
    </Container>
  </SC.ParticipationSection>
);

export default CourseParticipation;
