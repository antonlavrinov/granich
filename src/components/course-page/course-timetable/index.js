import React from "react";
import { Container } from "../../style";

import ArrowCurly from '../../../assets/svgs/arrow-curly.svg';
import ArrowMobile from '../../../assets/svgs/arrow-long-down.svg';
import TimetableImageGif from "../../../assets/images/graph-design/commitment/commitment.gif"
import LazyLoad from "react-lazyload"
import { graphql, useStaticQuery } from "gatsby";
import * as SC from "./style";
import CourseDontWorry from "../CourseDontWorry"
import { useLocalQuery } from "../../index/precourse-preparation/content-card/useLocalQuery";

const CourseTimetable = ({noGif, noKeepCalm, text, courseTitle, withCards}) => {
  const { behanceIcon, behanceLinkIcon } = useLocalQuery();

  const data = useStaticQuery(graphql`
    query timetableImage {
      imageTimetable: file(
        relativePath: { eq: "graph-design/timetable/timetable-image.jpg" }
      ) {
        childImageSharp {
          fluid(maxWidth: 700, quality: 100) {
            ...GatsbyImageSharpFluid
          }
        }
      }
      posterImageTimetable: file(
        relativePath: { eq: "poster-graphic-design/timetable-image.jpg" }
      ) {
        childImageSharp {
          fluid(maxWidth: 700, quality: 100) {
            ...GatsbyImageSharpFluid
          }
        }
      }
      foundationImageTimetable: file(
        relativePath: { eq: "foundation-of-graph-design/timetable-image.jpg" }
      ) {
        childImageSharp {
          fluid(maxWidth: 700, quality: 100) {
            ...GatsbyImageSharpFluid
          }
        }
      }
      metaphorsImageTimetable: file(
        relativePath: { eq: "metaphors-in-identity-page/timetable-image.jpg" }
      ) {
        childImageSharp {
          fluid(maxWidth: 700, quality: 100) {
            ...GatsbyImageSharpFluid
          }
        }
      }
    }
  `)

  const imageData = data.imageTimetable.childImageSharp.fluid

  const timetableImages = {
    "Графические метафоры": data.metaphorsImageTimetable.childImageSharp.fluid,
    "Фундамент Графдизайна": data.foundationImageTimetable.childImageSharp.fluid,
    "Плакатный Графдизайн": data.posterImageTimetable.childImageSharp.fluid,
  };

  return (
    <SC.TimetableSection>
      <Container>
        <SC.TimetableWrapper>
          <SC.TimetableTitle>
            Курс можно совмещать с работой.{" "}
            <span>Но расслабиться не получится!</span>
          </SC.TimetableTitle>
          <SC.TimetableInfoWrapper>
            <SC.TimetableInfoTextMobile>{text}</SC.TimetableInfoTextMobile>
            <SC.TimetableImageWrapper>
              <SC.TimetableImage fluid={noGif ? timetableImages[courseTitle] : imageData} />
            </SC.TimetableImageWrapper>

            <SC.TimetableInfoTextWrapper>
              <SC.TimetableInfoText>{text}</SC.TimetableInfoText>
              {!noGif && (
                <SC.TimetableInfoGifBlock>
                  <SC.TimetableGifWrapper>
                    <LazyLoad once>
                      <SC.TimetableGif src={TimetableImageGif} alt="commitment" />
                    </LazyLoad>
                  </SC.TimetableGifWrapper>
                  <SC.TimetableGifTextWrapper>
                    <SC.TimetableGifText>
                      Примерно с таким настроем нужно будет погрузиться в курс. Вы
                      придёте сюда чтобы развивать твёрдые навыки.
                    </SC.TimetableGifText>
                    <SC.TimetableGifPS>
                      P.S. Сцена из фильма Одержимость. Философия всего курса.
                    </SC.TimetableGifPS>
                  </SC.TimetableGifTextWrapper>
                </SC.TimetableInfoGifBlock>
              )}
            </SC.TimetableInfoTextWrapper>
          </SC.TimetableInfoWrapper>
          {!noKeepCalm && (
            <CourseDontWorry />
          )}
          {withCards && (
            <SC.BehanceContentCards>
              <ArrowCurly/>
              <ArrowMobile />
              <SC.ArrowText>
                Примеры готовых экзаменов
              </SC.ArrowText>
              {[
                {
                  imageUrl: "/images/poster-graphic-design/timetable-behance-cases/01.png",
                  link: "https://www.behance.net/gallery/142593515/redizajn-instagram-profilja",
                },
                {
                  imageUrl: "/images/poster-graphic-design/timetable-behance-cases/02.jpg",
                  link: "https://www.behance.net/gallery/141729021/redizajn-instagram-profilja-repetitora-po-obschestvoznaniju",
                },
                {
                  imageUrl: "/images/poster-graphic-design/timetable-behance-cases/03.jpg",
                  link: "https://www.behance.net/gallery/141394269/Content-redesign-for-the-theatre",
                },
                {
                  imageUrl: "/images/poster-graphic-design/timetable-behance-cases/04.png",
                  link: "https://www.behance.net/gallery/142241941/rjadom-%29",
                },
                {
                  imageUrl: "/images/poster-graphic-design/timetable-behance-cases/05.jpg",
                  link: "https://www.behance.net/gallery/141076167/Redesign-of-the-cyber-schools-instagram-profile",
                },
                // {
                //   imageUrl: "/images/poster-graphic-design/timetable-behance-cases/06.jpg",
                //   link: "https://www.behance.net/gallery/142537241/Farm-Bird-Food-Packaging-Design-and-Identity",
                // },
                {
                  imageUrl: "/images/poster-graphic-design/timetable-behance-cases/07.jpg",
                  link: "https://www.behance.net/gallery/141287597/redizajn-instagram-profilja",
                },
              ].map(({ imageUrl, link }) => (
                <SC.WrapperExternalLink
                    className="content"
                    behance
                    rel="noopener noreferrer"
                    href={link}
                    target="_blank"
                  >
                    <SC.Image src={imageUrl}></SC.Image>
                    <SC.Button>
                      <SC.ButtonText>Полный кейс</SC.ButtonText>
                      <SC.IconsWrapper>
                        <SC.BehanceIcon fluid={behanceIcon} />
                        <SC.BehanceOutIcon fluid={behanceLinkIcon} />
                      </SC.IconsWrapper>
                    </SC.Button>
                  </SC.WrapperExternalLink>
                )
              )}
            </SC.BehanceContentCards>
          )}
        </SC.TimetableWrapper>
      </Container>
    </SC.TimetableSection>
  )
};

export default CourseTimetable;
