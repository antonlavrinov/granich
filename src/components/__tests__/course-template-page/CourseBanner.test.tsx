import React from "react"
import { shallow } from "enzyme"
import { PureCourseBanner as CourseBanner } from "../../course-page/CourseBanner"

describe("<CourseBanner> component", () => {
  it("should render correctly", () => {
    const mockBoxImage = "imageString"

    expect(shallow(<CourseBanner box={mockBoxImage} />)).toMatchSnapshot()
  })
})
