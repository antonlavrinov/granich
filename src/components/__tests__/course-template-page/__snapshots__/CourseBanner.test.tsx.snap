// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseBanner> component should render correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <Styled(Component)
        fluid="imageString"
        style={
          Object {
            "position": "absolute",
          }
        }
      />
      <styled.div>
        <styled.div>
          <styled.div>
            <styled.div>
              Осознанная подборка по Графдизайну
            </styled.div>
          </styled.div>
          <styled.div>
            Это специальная подборка материалов для старта в графическом дизайне. Её необязательно изучать, чтобы пройти курс. Но если у вас есть время, то рекомендую. Это повысит качество вашего обучения на курсе.
          </styled.div>
        </styled.div>
        <styled.div>
          <styled.a
            href="https://granich.design/conscious-graphic-design-compilation"
            target="_blank"
          >
            Изучить
            <test-file-stub />
          </styled.a>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
